Bottom: 206218e7d615048485b790000210276d87523fe3
Top:    3bf1f85f839e36f2e139275bee8e8f5e0a7ad4d1
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-17 16:08:59 +0100

Don't adjust the Library command settings if they are NULL, or
an empty string.

(The empty string part is a fixup for
commit 31df66e87a810aac852139da9119b15c47fa8c71
 - this will teach me for bikeshedding)

The NULL part should fix bug lp-852598

Closes-bug: lp-852598


---

diff --git a/src/file.c b/src/file.c
index 8888432..c22f5c5 100644
--- a/src/file.c
+++ b/src/file.c
@@ -1417,7 +1417,8 @@ ReadLibraryContents (void)
    * skip this. This is used by default on Windows builds (set in main.c),
    * as we can't normally run shell scripts or expect to have m4 present.
    */
-  if (*Settings.LibraryContentsCommand != '\0')
+  if (Settings.LibraryContentsCommand != NULL &&
+      Settings.LibraryContentsCommand[0] != '\0')
     {
       /*  First load the M4 stuff.  The variable Settings.LibraryPath
        *  points to it.
diff --git a/src/main.c b/src/main.c
index ddd5675..d5a8487 100644
--- a/src/main.c
+++ b/src/main.c
@@ -1442,7 +1442,10 @@ REGISTER_ATTRIBUTES (main_attribute_list)
 {
   char *tmps;
 
-  if (Settings.LibraryCommand[0] != PCB_DIR_SEPARATOR_C && Settings.LibraryCommand[0] != '.')
+  if (Settings.LibraryCommand != NULL &&
+      Settings.LibraryCommand[0] != '\0' &&
+      Settings.LibraryCommand[0] != PCB_DIR_SEPARATOR_C &&
+      Settings.LibraryCommand[0] != '.')
     {
       Settings.LibraryCommand
 	=
@@ -1450,8 +1453,10 @@ REGISTER_ATTRIBUTES (main_attribute_list)
 		Settings.LibraryCommand,
 		NULL);
     }
-  if (Settings.LibraryContentsCommand[0] != PCB_DIR_SEPARATOR_C
-      && Settings.LibraryContentsCommand[0] != '.')
+  if (Settings.LibraryContentsCommand != NULL &&
+      Settings.LibraryContentsCommand[0] != '\0' &&
+      Settings.LibraryContentsCommand[0] != PCB_DIR_SEPARATOR_C &&
+      Settings.LibraryContentsCommand[0] != '.')
     {
       Settings.LibraryContentsCommand
 	=
